ppx.ml

```diff
@@ -0,0 +1,133 @@
+module Stre = Str
+open Parsetree
+open Ast_helper
+module AM = Ast_mapper
+
+let error ppf =
+  Format.kasprintf (fun s ->
+    let loc = !default_loc in
+    Exp.extension ~loc @@
+    AM.extension_of_error (Location.error ~loc s))
+    ppf
+
+let mklid m lid =
+  Exp.ident
+    {Location.loc = !default_loc ; txt = Longident.(Ldot (Lident m, lid))}
+let app m lid l =
+  Exp.(apply (mklid m lid)
+      (List.map (fun e -> Asttypes.Nolabel, e) l))
+
+let try_int s =
+  try `Int32 (Int32.of_string s)
```

Comment:
Yes, I though about that. We could definitely check if they are small enough to fit. I don't want to emit integers literals that are bigger than 31 bits though, as it's going to annoy people that do cross-compilation greatly.